app/src/components/nodeEntries/FormattedInput.tsx

```diff
@@ -1,26 +1,31 @@
 import { VStack, Text } from "@chakra-ui/react";
 import type { DatasetEntryInput } from "@prisma/client";
-import type { ChatCompletionMessage } from "openai/resources/chat";
-import FormattedMessage from "./FormattedMessage";
 import { FormattedJson } from "../FormattedJson";
+import { typedDatasetEntryInput } from "~/types/dbColumns.types";
 
 const FormattedInput = ({
-  input: { messages, tool_choice, tools },
+  input,
 }: {
   input: {
     messages: DatasetEntryInput["messages"];
     tool_choice?: DatasetEntryInput["tool_choice"];
     tools?: DatasetEntryInput["tools"];
   };
 }) => {
+  const { messages, tool_choice, tools } = typedDatasetEntryInput(input);
+
   return (
     <VStack spacing={8} maxW="full">
-      {(messages as unknown as ChatCompletionMessage[]).map((message, index) => (
+      {messages.map((message, index) => (
         <VStack key={index} alignItems="flex-start" w="full">
           <Text fontWeight="bold" color="gray.500">
             {message.role}
           </Text>
-          <FormattedMessage message={message} />
+          <Text whiteSpace="pre-wrap" maxW="full">
+            {typeof message.content === "string"
+              ? message.content
+              : JSON.stringify(message.content, null, 2)}
```

Comment:
We can work on making the display prettier in the future, but this is good enough for now!